README.md

```diff
@@ -380,6 +377,17 @@ configured with the following attributes:
   shown as a dropdown to the user.  It must match the list of port numbers in
   both order and number.
 
+You can also use the following nested XML tags:
+
+* `provider`: (Optional) A list of OpenStack providers configured in the
```

Comment:
Minor nit: this sounds like the xblock expects a _single_ nested `<provider>` tag, which is then interpreted as a list. I'd suggest to rephrase this as:

```markdown
* `provider`: Reference to an OpenStack provider configured in the platform.
  It `name` attribute must match one of the providers in the XBlock configuration,
  `capacity` specifies how many environments should be launched in that
  provider at maximum (where 0 means keep launching environments until 
  encountering a launch failure), and 
  `environment` specifies a content store path to a Heat environment file specifying
  template parameters for that provider.
  This element must be specified at least once, and can be specified multiple times
  to use one or more fallback providers.
```

The part where it says "where 0 means launch until encountering a launch failure" is of course subject to the rest of the discussion, i.e. what does setting 0 mean, as opposed to not defining the attribute at all. Is it equivalent (meaning 0 means launch until failure), or is it 0 actually 0 (meaning launch nothing here at all, something that one might use to temporarily disable a provider in a deployed course) .

Reply:
+1 for improving the wording.  As for 0 versus `None`, they're currently equivalent, in that they translate to infinite capacity.  However, you have a point:  we may want to use -1 for infinite, and 0 for disabled.  What do you think?